pyccel/parser/syntax/headers.py

```diff
@@ -59,15 +61,20 @@ def expr(self):
 class TemplateStmt(BasicStmt):
     """Base class representing a  template in the grammar."""
     def __init__(self, **kwargs):
-        self.dtype  = kwargs.pop('dtype')
+        self.dtypes  = kwargs.pop('dtypes')
         self.name   = kwargs.pop('name')
         BasicStmt.__init__(self)
 
     @property
     def expr(self):
-        l = [i.expr for i in self.dtype]
-        return Template(self.name, l)
+        if any(isinstance(d_type, FuncType) for d_type in self.dtypes):
+            msg = 'Functions in a template are not supported yet'
+            errors.report(msg,
+                        severity='fatal')
```

Comment:
Can you add a test in tests/errors to make sure that this is correctly raised please